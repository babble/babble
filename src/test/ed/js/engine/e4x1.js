
function myprint( s ){
    if ( ! s )
        return print( s );
    s = s.toString();
    s = s.replace( /[\r\n\s]+/g , "" );
    print( s );
}

var sales = <sales vendor="John">
    <name>John Smith</name>
    <silly><foo>a</foo></silly>
    <item type="peas" price="4" quantity="6"/>
    <item type="carrot" price="3" quantity="10"/>
    <item type="chips" price="5" quantity="3"/>
    <zzz a="5" />
    <zzz a="6" />
    <zzz a="7" />
    </sales>;

myprint( sales );
myprint( sales.@vendor );
myprint( sales.silly );
myprint( sales.name );

myprint( sales.zzz );
myprint( sales.item.length() );


x = XML( "<foo><a>6</a></foo>" );
print( x.a );

x = new XML( "<foo><a>6</a></foo>" );
print( x.a );
