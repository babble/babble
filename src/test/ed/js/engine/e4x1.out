<salesvendor="John"><name>JohnSmith</name><silly><foo>a</foo></silly><itemtype="peas"price="4"quantity="6"/><itemtype="carrot"price="3"quantity="10"/><itemtype="chips"price="5"quantity="3"/></sales>
John
<silly><foo>a</foo></silly>
JohnSmith
3
